bot/cogs/help.py

```diff
@@ -0,0 +1,234 @@
+from inspect import Parameter
+import textwrap
+import typing as t
+
+import discord
+from discord.ext import commands, menus
+import discord.utils
+
+
+class HelpSource(menus.ListPageSource):
+    """The Help menu."""
+
+    def __init__(
+            self,
+            signature: t.Callable,
+            filter_commands: t.Coroutine,
+            prefix: str,
+            author: discord.User,
+            cogs: dict
+    ) -> None:
+        """Create the menu."""
+        self.get_command_signature = signature
+        self.filter_commands = filter_commands
+        self.prefix = prefix
+        self.menu_author = author
+        super().__init__(
+            [(cog, cogs[cog]) for cog in sorted(
+                cogs,
+                key=lambda cog: cog.qualified_name if cog else "ZZ"
+            )
+            ],
+            per_page=1,
+        )
+
+    async def format_page(
+            self,
+            menu: menus.Menu,
+            cog_tuple: tuple,
```

Comment:
You can specify this type in more detail with [`t.Tuple`](https://docs.python.org/3/library/typing.html#typing.Tuple)

Reply:
I'm not sure it's really useful. More like adding code just to add code

Reply:
Same as before : reaaaaaaly long tuple typehint, serving no purpose

Reply:
defining this with typehints would provide everyone with a clear understanding of what arguments this function takes, which is necessary to make further editing easy for others if you don't like the idea of type hints, in theory, you could specify it in a longer docstring:
```py
"""
Main function description

[Optional more detailed description]
* menu: `menus.Menu`:
    [Optional argument description]
"""
```
but using type hints seems a lot more simple and that way the editors will also know what's in that tuple which means it will easily provide autocompletion when working with it. It's what type hints are for after all, and as stated in the comment on a similar requested change, you can import some classes directly instead of doing `commands.X` which would make those type hints much shorter

Reply:
The thing is, this method is used only once in a specific situation. It was NOT made to be reused (we won't have 100 places where we use help). So there's no real need to make a super specific typehint, since it won't be reused somewhere else

Reply:
that may be, but if someone would need to redesign help without rewriting it, using this type hint will make it so much easier to recognize the purpose of that argument in a function, also I'm not just talking about this specific function, this applies to all functions with incomplete type hints. Even as a 1-time use function, there is no reason to not follow the standards we have everywhere else and as I said, these type hints won't be that long if you import some objects directly:
```py
t.Tuple[t.Optional[Cog], t.List[Command]]
```

Reply:
It's in the latest commit already